types/list-stream/index.d.ts

```diff
@@ -7,16 +7,21 @@
 
 import { Duplex, DuplexOptions } from "stream";
 
-declare namespace ListStream {
+interface ListStreamMethod {
+    (callback?: (err: Error, data: any[]) => void): ListStream;
+    (options?: DuplexOptions, callback?: (err: Error, data: any[]) => void): ListStream;
 }
 
-declare class ListStream extends Duplex {
-    constructor(callback?: (err: Error, data: any[]) => void);
-    constructor(options?: DuplexOptions, callback?: (err: Error, data: any[]) => void);
+interface ListStreamConstructor extends ListStreamMethod {
```

Comment:
@IanStorm  I'm surprised you moved away from using a class here. Was there a particular reason that you thought a new-able constant would be better?

Reply:
I see, that makes sense!

I do have a recommendation regarding your "import style" point.
Under the ES6 spec, it is not correct to expect something imported with `import * as foo from "bar";` to be callable or newable. If consumers of this package want to use ES6-style imports in TypeScript, they should use `import ListStream from "list-stream";` and the `--esModuleInterop` compilation flag. You can do the same in your tests by adding that compilation option to the tsconfig.json in this folder.

See [here](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#support-for-import-d-from-cjs-form-commonjs-modules-with---esmoduleinterop) and [here](https://github.com/definitelyTyped/DefinitelyTyped/#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports) for more information.

Reply:
Thank you for the recommendation. The links you provided were really helpful, not only with respect to this merge request, but also to improve my general understanding. :)
I pushed new commits (f368a2dbfdd73511fbde3ab961ec111ce0486646 and 3f185078d8bdf57c6669b08a5ea135de65555d9d) to include the required changes.

Keeping in mind that my goal was to support "import style" and "require style" I was thinking that the "tests" file should also reflect that. Currently it only reflects the "import style". So what would you say about having two `list-stream-tests.ts` files, i.e. one for "import style" renamed to (for example) `list-stream-tests-import.ts`, and another for "require style" named (for example) `list-stream-tests-require.ts`? Do you think it is a good idea? Is there any convention on how to name or create the test files?

Reply:
There definitely does need to be a `list-stream-tests.ts` file. However, you can make a `tests` directory and add your two test files there, then import them into `list-stream-tests.ts`. Alternatively, you can use both imports in a single file:
```ts
import ListStream = require("list-stream");
import ListStream2 from "list-stream";
import { obj } from "list-stream";
```